pylot/planning/frenet_optimal_trajectory/frenet_optimal_trajectory.py

```diff
@@ -0,0 +1,310 @@
+
+    return True
+
+
+def _check_paths(fplist, ob):
+    okind = []
+    for i, _ in enumerate(fplist):
+        # Max speed check
+        if np.any(np.abs(fplist[i].s_d) > MAX_SPEED):
+            continue
+        # Max accel check
+        elif np.any(np.abs(fplist[i].s_dd) > MAX_ACCEL):
+            continue
+        # Max curvature check
+        elif np.any(np.abs(fplist[i].c) > MAX_CURVATURE):
+            continue
+        # Collision check
+        elif not _check_collision(fplist[i], ob):
+            continue
+
+        okind.append(i)
+
+    return [fplist[i] for i in okind]
+
+
+def frenet_optimal_planning(csp,
+                            s0,
+                            c_speed,
+                            c_d,
+                            c_d_d,
+                            c_d_dd,
+                            ob,
+                            target_speed):
+    """Find the frenet optimal trajectory.
+
+    Args:
+        csp (:py:class:`planning.control.mpc.utils.CubicSpline2D`): Cubic
+            spline defining the intended trajectory to follow
+        s0 (:float:): s-position along the spline
+        c_speed (:float:): the speed at which the vehicle is moving
+        c_d (:float:): the lateral offset d from the frenet frame
+        c_d_d (:float:): the lateral speed of the vehicle
+        c_d_dd (:float:): the lateral acceleration of the vehicle
+        ob (:list([x, y]:): list of obstacle origins
+        target_speed (:float:): target speed to reach
+
+    Returns:
+        bestpath
+            (:py:class:`~pylot.planning.frenet_optimal_trajectory.FrenetPath`):
+            frenet optimal trajectory
+    """
+    fplist = _calc_frenet_paths(c_speed, c_d, c_d_d, c_d_dd, s0, target_speed)
+    fplist = _calc_global_paths(fplist, csp)
+    fplist = _check_paths(fplist, ob)
+
+    # find minimum cost path
+    mincost = float("inf")
+    bestpath = None
+    for fp in fplist:
+        if mincost >= fp.cf:
+            mincost = fp.cf
+            bestpath = fp
+
+    return bestpath
+
+
+def generate_target_course(x, y):
+    """
+    Return the global course and frenet path defined by a list of waypoints.
+
+    Args:
+        x (:list(float):): list of x positions [m]
+        y (:list(float):): list of y positions [m]
+
+    Returns:
+        rx (:list(float):): list of x positions [m]
+        ry (:list(float):): list of y positions [m]
+        ryaw (:list(float):): list of yaws [rads]
+        rk (:list(float):): list of curvatures [1/m]
+        csp (:py:class:`~pylot.control.mpc.utils.CubicSpline2D`): Cubic spline
+            defined by input x, y
+    """
+    csp = CubicSpline2D(x, y)
+    s = np.arange(0, csp.s[-1], 0.1)
+
+    rx, ry, ryaw, rk = [], [], [], []
+    for i_s in s:
+        ix, iy = csp.calc_position(i_s)
+        rx.append(ix)
+        ry.append(iy)
+        ryaw.append(csp.calc_yaw(i_s))
+        rk.append(csp.calc_curvature(i_s))
+
+    return rx, ry, ryaw, rk, csp
+
+
+def main():
```

Comment:
Can you please remove main?

Reply:
@ICGog I'd prefer to keep this as a debugging tool in case needed in the future, would you prefer if I moved it to it's own file?